src/features/Nav/Navbar.tsx

```diff
@@ -61,13 +62,26 @@ export default class Navbar extends React.Component<
         confirmTime: new Date().toString(),
       },
       // hasSponsorInfo: false,
+      hasBorder: false,
     };
     this.checkLoggedIn();
   }
 
+  listenToScrollEvent = (): void => {
+    document.addEventListener('scroll', () => {
+      requestAnimationFrame(() => {
+        this.calculateScrollDistance();
+      });
+    });
+  };
+
+  calculateScrollDistance = (): void => {
+    this.setState({ hasBorder: window.pageYOffset !== 0 });
+  };
+
   public async componentDidMount() {
     let hacker;
-
+    this.listenToScrollEvent();
```

Comment:
Can you unsubscribe from this listener in `componentWillUnmount`? (see https://stackoverflow.com/a/38564123)

Reply:
yup thanks!